src/hotspot/share/gc/shenandoah/heuristics/shenandoahOldHeuristics.cpp

```diff
@@ -310,37 +309,29 @@ void ShenandoahOldHeuristics::prepare_for_old_collections() {
 
   const size_t num_regions = heap->num_regions();
   size_t cand_idx = 0;
-  size_t total_garbage = 0;
   size_t immediate_garbage = 0;
   size_t immediate_regions = 0;
   size_t live_data = 0;
 
   RegionData* candidates = _region_data;
   for (size_t i = 0; i < num_regions; i++) {
     ShenandoahHeapRegion* region = heap->get_region(i);
-    if (!_old_generation->contains(region)) {
+    if (!region->is_old()) {
       continue;
     }
 
     size_t garbage = region->garbage();
     size_t live_bytes = region->get_live_data_bytes();
-    total_garbage += garbage;
     live_data += live_bytes;
 
-    // Only place regular regions into the candidate set
-    if (region->is_regular()) {
```

Comment:
I understand that the rationale for the PR is to make sure we do not overlook pinned regions.  What I'm not seeing is how the original code is excluding the pinned regions from being inserted into the "old collection set".  To help my review, can you point me to that part of the original code?

Reply:
No need for this.  I now understand that is_regular() does not include is_pinned_regular().